### DISCUSSION TITLE
Doesn't work anymore

### DISCUSSION THREAD

**Discussion by Lionel90370** [moraroy/NonSteamLaunchers-On-Steam-Deck#118] [Q&A] [answered]
Hello

1st I'd like to thank the author of NonSteamLauncher for his work, It is a great tool.

About a month ago, I doscovered this script and it worked wonderfully with the installation of some launchers.
Lately I wanted to add the EA app launcher to my steamdeck, just before doing so I came here to check if there was a new version, which I found and downloaded.

I launched it, noticed that the paths for the allready installed ones weren't showing anymore, but did not care more than that.
I selected the EA app, and proceeded with the installation, but for some reasons it didn't work at all.
I retried with the start fresh button, same result.
Later, I found that the other launchers were messed up, and wouldn't work at all.
I tried to re install them all (Uplay, GoG, Battlenet, Epic, EA app, all checked), removed them from my steam library, but the script was stuck somewhere during the GoG download.
I stoped it after something like 1 hour with no progress.
But it looks like it installed nothing (at least nothing shows in the steam interface).
I can't say if the previously installed files are still on my Deck. (so if there's a blocking here)

May I ask for some helpfull advices ?
(sorry for bad gramma, english isn't my native language)

You can find the log file attached.

Thank you.
[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/12344130/NonSteamLaunchers-install.log)

**Comment by moraroy** [comment]
@Lionel90370 just updated the .sh file and delted a line of code please try it again and let me know!

**Comment by Lionel90370** [comment]
Sadly, it gave me the same result.
I don't know if I did anything wrong.

Log attached.
[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/12369159/NonSteamLaunchers-install.log)

**Comment by mrmldeboy** [comment]
All of my launchers stopped working after removing my old version and re-installing with current version. Trying to download a previous version of the installer doesn't work as it simply updates to v2.99 during execution. Not sure why this changed as everything worked perfectly in v2.96

**Comment by Lionel90370** [comment]
I tried to give it another go, with a different launcher than the ones I allready mentioned.

This time I selected "Hulu" and ran the script.
It works !! Hulu is now showing in my Steam Librabry.


Then I relaunched the .desktop, but in the list there is no path for Hulu.

From my previous experience, I suppose that if I hit start again the script, hit the "start fresh button", then porceed for another installation of Hulu, it won't show in Steam library.

So I also suppose the script would work only 1 time, I can think of 2 solutions:
- "manually" install the launcher I wish
-  reinstall steam OS and run the script for the desired lanchers.
Just would it work wih updates ? I hope so, just I'm not very sure.

By the way, are you expecting an updated version in a near future ?
Log attached
[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/12905421/NonSteamLaunchers-install.log)

**Reply by Lionel90370** [reply]
Some update after the latest fix for Ubisoft connect.
It now works with Ubisoft connect, which I could install again. Many thanks for your work.

However ,the other launchers I am looking for weren't the same story.
I tried to install those 1 by 1.
- Epic games and GoG galaxy: the script looked stuck during the download process, I stoped it after some time.
- Battle.net and EA app: the script seems to work fine, but no entries are appearing in the steam library.

Here are my logs (1 for each) 
[NonSteamLaunchers-install EAapp.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/13312495/NonSteamLaunchers-install.EAapp.log)
[NonSteamLaunchers-install epic.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/13312496/NonSteamLaunchers-install.epic.log)
[NonSteamLaunchers-install gog.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/13312497/NonSteamLaunchers-install.gog.log)
[NonSteamLaunchers-instal battlel.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/13312498/NonSteamLaunchers-instal.battlel.log)

**Comment by Lionel90370** [comment]
It looks like the 3.5.5 Steam OS update broke everything.

**Reply by moraroy** [reply, marked as answer]
@Lionel90370 I just updated to 3.0 you can try it now let me know

**Reply by Lionel90370** [reply]
Everything is working now.
Thank you
